src/converter.ts

```diff
@@ -501,10 +509,30 @@ export class ProtocolToMonacoConverter {
         const contents = Array.isArray(hover.contents) ? hover.contents : [hover.contents];
         const range = this.asRange(hover.range)!;
         return {
-            contents, range
+            contents: this.asIMarkdownStrings(contents),
+            range: range
+        };
+    }
+
+    asIMarkdownString(string: MarkedString): monaco.IMarkdownString {
```

Comment:
you can use `typeof` to avoid down casting to `any`:
```ts
asIMarkdownString(string: MarkedString): monaco.IMarkdownString {
        if (typeof string === 'string') {
            return {
                value: string
            }
        }
        const { language, value } = string;
        return {
            value: '```' + language + '\n' + value + '\n```'
        };
    }
```

Reply:
I've replaced my code with your suggested snippet, thanks!